frame/support/test/tests/derive_no_bound.rs

```diff
@@ -128,9 +146,18 @@ fn test_struct_unnamed() {
 	let b = StructUnnamed::<Runtime, ImplNone, ImplNone>(1, 2, 4, Default::default());
 
 	assert!(b != a_1);
+	assert!(b > a_1);
```

Comment:
I see that the existing testing is rather poor. Ideally we want some UI tests to check success and error cases (path would be here https://github.com/paritytech/substrate/blob/237874bb15bbac174b6cb8d594053b5512f32ca2/frame/support/test/tests/derive_no_bound_ui/partial_eq.rs#L10-L11).